r/tests/testthat/test-dataset.R

```diff
@@ -1767,3 +1767,13 @@ test_that("Dataset writing: unsupported features/input validation", {
     write_dataset(ds, tempfile(), basename_template = NULL)
   )
 })
+
+# see https://issues.apache.org/jira/browse/ARROW-11328
+test_that("Collecting zero columns from a dataset doesn't return entire dataset", {
+  tmp <- tempfile()
```

Comment:
Since this test writes and reads a Parquet file, we need to skip it if the Arrow C++ library was not built with Parquet
```suggestion
  skip_if_not_available("parquet")
  tmp <- tempfile()
```